src/engine/SCons/Tool/tex.py

```diff
@@ -297,8 +297,8 @@ def check_MD5(filenode, suffix):
         logfilename = targetbase + '.log'
         logContent = ''
         if os.path.isfile(logfilename):
-            with open(logfilename, "r") as f:
-                logContent = f.read()
+            with open(logfilename, "rb") as f:
+                logContent = f.read().decode(errors='replace')
```

Comment:
If it's not utf-8 we shouldn't be decoding it as such, this just seems like it's papering over the problem?

Reply:
I agree. But how do we know which encoding is used? Probably it depends on the fontenc options in the source. We'd need someone who understands exactly what Latex is doing here.

My solution isn't nice. But at the moment the build fails in such a case. And since the character replacements only affect a piece of text from the document that is part of a warning message, I find it acceptable. If anybody comes up with a thorough solution later, all the better.

Reply:
I'll tag our resident LaTEX expert..  @managan can you take a look?

Reply:
You can also look at it that way: when you run Latex manually (i.e. not from SCons), your shell will do the same replacement to the logs that Latex writes to stdout (since most distributions nowadays use UTF-8 encoding by default for the shell). Handling this within SCons seems like being more catholic than the pope. This is a question of resilience. Imagine that Bash crashes when you run the test case!